src/state/trajectory/conversion-data-types.ts

```diff
@@ -36,5 +36,8 @@ export type TemplateMap = {
 };
 
 export interface Template {
-    [key: string]: CustomType
-}
\ No newline at end of file
+    [key: string]: CustomType;
+}
+export interface ExtensionMap {
+    [key: string]: string;
+}
```

Comment:
once you change this to 
```TypeScript
export type ExtensionMap = {
    [key in AvailableEngines]: string;
}
```
this looks good to me

Reply:
Done! Had to add "txt" to validFileExtensions for all four engines, I don't actually know what file extensions they each use.